### ISSUE TITLE
pre insert/post insert

### ISSUE THREAD

**Issue by ronoc** [seiflotfy/chronos#17]
should this not be pre operation/post operation

It's not just inserts we are interested in. Should this not be made more generic such as each event being

- operation type (put, delete, update etc)
- id (what entity was affected)
- time (when it was carried on)
- source (who caused it)

**Comment by martinpinto** [comment]
I agree, it makes more sense to rename the plugin system to make it more generic. At the time we only support the add operation, but we need to take all other operation types into consideration.

**Comment by seiflotfy** [comment]
This is an event log. the only operation that will happen is a put and very seldomly a delete (we will drop indices by default). There won't be an update since updating an event is manipulating history, which I am not a fan of. If something should not be in the DB then the developer needs to make sure its removed blocked from being inserted in the DB (via a pre insert) if possible.
